File: ambeth/Ambeth.IoC/ambeth/ioc/exception/LinkException.cs
using De.Osthus.Ambeth.Ioc.Link;
using System;

namespace De.Osthus.Ambeth.Ioc.Exceptions
{
    public class LinkException : System.Exception
    {
        protected readonly AbstractLinkContainer linkContainer;

        public LinkException(String message, System.Exception cause, AbstractLinkContainer linkContainer)
            : base(message, cause)
        {
            this.linkContainer = linkContainer;
        }

        public LinkException(String message, AbstractLinkContainer linkContainer)
            : base(message)
        {
            this.linkContainer = linkContainer;
        }
    }
}

File: jambeth/jAmbeth.IoC/src/main/java/de/osthus/ambeth/ioc/exception/LinkException.java
package de.osthus.ambeth.ioc.exception;

import de.osthus.ambeth.ioc.link.AbstractLinkContainer;

@SuppressWarnings("serial")
public class LinkException extends RuntimeException
{
	transient protected final AbstractLinkContainer linkContainer;

	public LinkException(String message, Throwable cause, AbstractLinkContainer linkContainer)
	{
		super(message, cause);
		this.linkContainer = linkContainer;
	}

	public LinkException(String message, AbstractLinkContainer linkContainer)
	{
		super(message);
		this.linkContainer = linkContainer;
	}
}
